master/pkg/model/task_config_policy.go

```diff
@@ -20,6 +24,43 @@ const (
 	NTSCType WorkloadType = "NTSC"
 )
 
+// ExperimentTaskConfigPolicies is the bun model of a task config policy.
+type ExperimentTaskConfigPolicies struct {
+	bun.BaseModel   `bun:"table:task_config_policies"`
+	WorkspaceID     *int                     `bun:"workspace_id"`
+	LastUpdatedBy   UserID                   `bun:"last_updated_by,notnull"`
+	LastUpdatedTime time.Time                `bun:"last_updated_time,notnull"`
+	WorkloadType    WorkloadType             `bun:"workload_type,notnull"`
+	InvariantConfig expconf.ExperimentConfig `bun:"invariant_config"`
+	Constraints     Constraints              `bun:"constraints"`
+}
+
+// NTSCTaskConfigPolicies is the bun model of a task config policy.
+type NTSCTaskConfigPolicies struct {
+	bun.BaseModel   `bun:"table:task_config_policies"`
+	WorkspaceID     *int          `bun:"workspace_id"`
+	LastUpdatedBy   UserID        `bun:"last_updated_by,notnull"`
+	LastUpdatedTime time.Time     `bun:"last_updated_time,notnull"`
+	WorkloadType    WorkloadType  `bun:"workload_type,notnull"`
+	InvariantConfig CommandConfig `bun:"invariant_config"`
+	Constraints     Constraints   `bun:"constraints"`
+}
+
+// ResourceConstraints are non-overridable resource constraints.
+// Submitted workloads that request resource quanities exceeding defined resource constraints in a
+// given scope are rejected.
+type ResourceConstraints struct {
+	MaxSlots *int `json:"max_slots"`
+}
+
+// Constraints are non-overridable workload constraints.
+// Submitted workloads whose config's respective field(s) exceed defined constraints within a given
+// scope are rejected.
+type Constraints struct {
+	ResourceConstraints *ResourceConstraints `json:"resources"`
+	PriorityLimit       *int                 `json:"priority_limit"`
+}
+
```

Comment:
nit:  don't object to `ResourceConstraints` and `Constraints` being in `package model`, but I am curious why they are here while `NTSCConfigPolicies` is in package `configpolicy`?

Reply:
`ResourceConstraints` and `Constraints` were initially part of package `configpolicy`. I removed them because of an import cycle. `NTSCConfigPolicies` needs `model.CommandConfig` for the invariant config, and if we add `Constraints` to the `configpolicy` package, then `model.NTSCTaskConfigPolicies` will need `configpolicy.Constraints`

Reply:
Oh .... makes sense. Thanks for explaining! I thought the import cycle was between `configpolicy` and `db`, not `model`.

Reply:
Np! 😄